metals/src/main/scala/scala/meta/internal/metals/doctor/Doctor.scala

```diff
@@ -531,8 +634,38 @@ final class Doctor(
     "Try removing the directories .metals/ and .bloop/, then restart metals And import the build again."
   private val buildServerNotResponsive =
     "Build server is not responding."
+  private val dateTimeFormat = new SimpleDateFormat("dd MMM HH:mm:ss")
 }
 
 case class DoctorVisibilityDidChangeParams(
     visible: Boolean
 )
+
+object Doctor {
+
+  /**
+   * Extracts the short summary [[scala.meta.internal.metals.Report.shortSummary]]
+   * from an error report file.
+   * @param file error report file
+   * @param root workspace folder root
+   * @return short summary or nothing if no summary or error while reading the file
+   */
+  def getErrorReportSummary(
+      file: TimestampedFile,
+      root: AbsolutePath,
+  ): Option[String] = {
+    def decode(text: String) =
+      text.replace(StdReportContext.WORKSPACE_STR, root.toString())
+    for {
+      lines <- Try(Files.readAllLines(file.toPath).asScala.toList).toOption
+      reversed = lines.reverse
+      index = reversed.indexWhere(_.startsWith(Report.summaryTitle))
+      if index >= 0
+    } yield reversed
+      .slice(0, index)
+      .reverse
+      .dropWhile(_ == "")
+      .map(decode)
+      .mkString("\n")
```

Comment:
```suggestion
      index = lines.indexWhere(_.startsWith(Report.summaryTitle))
      if index >= 0
    } yield lines
      .slice(index, lines.length)
      .takeWhile(_ != "")
      .map(decode)
      .mkString("\n")
```
do we need reverse at all?

Reply:
You can also use `lastIndexWhere` if needed.